### PR TITLE
validator: add more Alternator index creation tests

### PR DESCRIPTION
**Pull request by m-szymon**
Extends Alternator index creation tests, verifying behaviour in several edge cases, especially validation of input paramteres.

Follow-up #452

### CODE REVIEW

File: crates/validator/src/alternator/create_table.rs [hunk 1]
```diff
@@ -192,6 +194,278 @@ async fn create_describe_and_delete_table_with_vector_index(actors: TestActors)
...
+    alternator::delete_table(&client, &table_name).await;
+
+    alternator::wait_for_no_index(&vs_clients, &lower_index).await;
+    alternator::wait_for_no_index(&vs_clients, &upper_index).await;
+
+    info!("finished");
+}
+
+/// Index names are scoped to a CQL keyspace (`alternator_<table>`), so the
+/// same index name on case-distinct tables should be independent.
```
Comment (QuerthDP) [inline crates/validator/src/alternator/create_table.rs:253]: Do I understand correctly that each table has its own keyspace in alternator?
And having that assumption we could create same-named indexes on different tables because they're in different keyspaces. If we had one keyspace in such case it would error out like in CQL. Is it possible to have one keyspace but two tables in Alternator?

**Review by QuerthDP — APPROVED** [review]
Looks good. 
I've left one comment but AFAIK it's a correct assumption, just curious.

File: crates/validator/src/alternator/create_table.rs (hunk 1 above)
Comment (m-szymon) [reply crates/validator/src/alternator/create_table.rs:253]: > Do I understand correctly that each table has its own keyspace in alternator?

Yes. DynamoDB doesn't have concept o keyspace. In Alternator we use it internally for cases exactly like this.

> Is it possible to have one keyspace but two tables in Alternator?

In Alternator keyspace we have other "special" tables - MV, CDC..., but these are not accessible as basic tables. There can be only one "base" table.
